echopype/echodata/combine.py

```diff
@@ -435,14 +446,296 @@ def _check_echodata_channels(
+        all_chan_ds_list = [ed[ed_group] for ed in eds]
+
+        # select only the appropriate channels from each Dataset
+        ds_list = [
+            ds.sel(channel=ed_group_chan_sel[ed_group])
+            if ed_group_chan_sel[ed_group] is not None
+            else ds
+            for ds in all_chan_ds_list
+        ]
+
+        if ds_list:
+            # Get all of the keys and attributes
+            ds_attrs = [ds.attrs for ds in ds_list]
+            # Attribute holding
+            attrs_dict[ed_group] = ds_attrs
+
+            # Checks for ascending time in dataset list
+            _check_ascending_ds_times(ds_list, ed_group)
+
+            # get all dimensions in ds that are append dimensions
+            ds_append_dims = set(ds_list[0].dims).intersection(APPEND_DIMS)
+
+            # Checks for filter parameters for "Vendor_specific" ONLY
+            if ed_group == "Vendor_specific":
+                _check_filter_params(ds_list, ed_group, ds_append_dims)
+
+            if len(ds_append_dims) == 0:
+                combined_ds = ds_list[0]
+            else:
+                combined_ds = xr.Dataset()
+                for dim in ds_append_dims:
+                    drop_dims = [c_dim for c_dim in ds_append_dims if c_dim != dim]
+                    sub_ds = xr.concat(
+                        [ds.drop_dims(drop_dims) for ds in ds_list],
+                        dim=dim,
+                        coords="minimal",
+                        data_vars="minimal",
+                    )
+                    combined_ds = combined_ds.assign(sub_ds.variables)
+
+            # Modify default attrs
+            if ed_group == "Top-level":
+                ed_group = "/"
+
+            # Merge attributes and set to dataset
+            group_attrs = _merge_attributes(ds_attrs)
+
+            # Empty out attributes for now, will be refilled later
+            combined_ds.attrs = group_attrs
+
+            # Add combined flag and update conversion time for Provenance
+            if ed_group == "Provenance":
+                combined_ds.attrs.update(
+                    {
+                        "is_combined": True,
+                        "conversion_time": datetime.datetime.utcnow().strftime(
+                            "%Y-%m-%dT%H:%M:%SZ"
+                        ),
+                    }
+                )
+
+            # Data holding
+            tree_dict[ed_group] = combined_ds
+
+    # Capture provenance for all the attributes
+    prov_ds = _capture_prov_attrs(attrs_dict, echodata_filenames, sonar_model)
+    # Update the provenance dataset with the captured data
+    prov_ds = tree_dict["Provenance"].assign(prov_ds)
+    # Update filenames to iter integers
+    prov_ds[FILENAMES] = prov_ds[FILENAMES].copy(data=np.arange(*prov_ds[FILENAMES].shape))
+    tree_dict["Provenance"] = prov_ds
+
+    return tree_dict
+
+
 def combine_echodata(
     echodatas: List[EchoData] = None,
```

Comment:
How about changing `echodatas` to `echodata_list`? I couldn't stand adding s to data, and it is better to be explicit about it needs a list.
```suggestion
    echodata_list: List[EchoData] = None,
```